eventstore/storethread.go

```diff
@@ -62,27 +77,28 @@ func (a *singleThreadAdapter) Start() {
 	go a.threadToStore(&wg)
 	go a.storeToThread(&wg)
 	wg.Wait()
+	a.goRoutines.Add(2)
 }
 
 func (a *singleThreadAdapter) threadToStore(wg *sync.WaitGroup) {
+	defer a.goRoutines.Done()
 	sub := a.api.Subscribe(tservopts.ThreadID(a.threadID))
 	defer sub.Discard()
 	wg.Done()
 	for {
 		select {
-		case <-a.ctx.Done():
-			log.Infof("cancelling thread event dispatch to store for thread %s since ctx was cancelled", a.threadID)
+		case <-a.closeChan:
+			log.Debug("closing thread-to-store flow on thread %s", a.threadID)
 			return
 		case rec, ok := <-sub.Channel():
 			if !ok {
-				log.Info("ending dispatch of events to store since thread channel was closed") // ToDo: reconsider action
+				log.Errorf("notification channel closed, not listening to external changes anymore")
```

Comment:
Switched to `Error` since this shouldn't be OK. Tempted to `Fatal`, since current result would mean that no more flow of events will happen from Threads to Store. (For some reason Threads closed our listener...)